content/articles/introduction-to - application-performance-management/index.md

```diff
@@ -0,0 +1,88 @@
+
+- Business-wise, it reduces operational costs by eliminating manual monitoring of application performance. This, plus increased efficiency of computer programs, boosts revenue production and results in higher profits for a company.
+
+### Challenges faced by APM
+Current data management systems are built sophisticated enough to offer more than just computer resource utilization reports. For example, they can single out individual transactions from a pool of highly distributed data systems and monitor them. However, this sophistication is limited by high data rates. This is because, for ulterior analysis, data demands extended periods of storage time. This results in large data that tends to overload APM tools. This limitation is addressed with modern key-value data stores developed to accommodate large data while allowing viewing of up-to-date computer infrastructure.
+
+The rising development of cloud-native applications is also a major challenge for APM. The applications generate huge amounts of telemetry data. This makes it difficult to locate specific events within an application's infrastructure. To tackle this challenge, companies combine management tools to monitor individual components within an application.
+
+### Tools of APM
+Tools of APM are programs used to monitor the performance of applications. They are selected based on the users' preferences. These preferences range from in-depth business transaction monitoring, solid analysis ability to costs of the monitoring tool. The data size in the application being managed is also an important aspect to consider when choosing a monitoring tool. For instance, [Stackify Retrace](https://consoltech.com/blog/blacklisting-vs-whitelisting/) is a widely used tool preferred by software developers. However, it is relatively pricey at $79 per month as of June 2021. [Stagemonitor](https://www.stagemonitor.org/), on the other hand, is free. It is used by developers using Java. Small business owners prefer [Datadog](https://www.geeksforgeeks.org/network-address-translation-nat/#:~:text=Network%20Address%20Translation%20(NAT)%20is,access%20to%20the%20local%20hosts.) and [Scouter](https://kubernetes.io/docs/concepts/services-networking/service/). These owners have to part with $31 monthly to use Datadog, in contrast with the free Scouter.
+
+WordPress businesses prefer [New Relic](https://newrelic.com/ ), which charges $99 each month as of June 2021, while [ITRS](https://en.wikipedia.org/wiki/Semaphore_(programming)) is the tool of management for most banking entities. [Pinpoint](https://developers.redhat.com/blog/2018/10/22/introduction-to-linux-interfaces-for-virtual-networking#:~:text=A%20Linux%20bridge%20behaves%20like,VLAN%20filter%2C%20and%20multicast%20snooping.), [AppDynamics](https://searchnetworking.techtarget.com/definition/routing-table) and [Dynatrace](https://www.dynatrace.com/ ) are powerful monitoring tools and are used by large enterprises.
```

Comment:
Noticed that some of the links don't lead to the corresponding tools in this section.

Reply:
an oversight on my part. This is now resolved @collins-a